### PR TITLE
Add bash-style console shortcuts

### PR DESCRIPTION
**Pull request by matthew-mcallister**
A couple bugs had to be fixed. There are a couple minor tangential changes as well that improve code consistency.

### CODE REVIEW

**Comment by raa-eruanna** [comment]
It might be helpful if you gave a list of the new keyboard shortcuts.

**Comment by matthew-mcallister** [comment]
Sure. It's only a subset of the existing shortcuts (and certainly more could be added), so here they are pulled from the man page:
```
    Commands for Moving
       beginning-of-line (C-a)
              Move to the start of the current line.
       end-of-line (C-e)
              Move to the end of the line.
       forward-word (M-f)
              Move forward to the end of the next word.  Words  are  composed  of
              alphanumeric characters (letters and digits).
       backward-word (M-b)
              Move  back  to  the start of the current or previous word.  Words are composed
              of alphanumeric characters (letters and digits).

   Killing and Yanking
       kill-line (C-k)
              Kill the text from point to the end of the line.
       unix-line-discard (C-u)
              Kill  backward from point to the beginning of the line.  The killed text is
              saved on the kill-ring.
       unix-word-rubout (C-w)
              Kill  the  word  behind  point, using white space as a word boundary.  The
              killed text is saved on the kill-ring.
       yank (C-y)
              Yank the top of the kill ring into the buffer at point.
```

**Comment by raa-eruanna** [comment]
So far, testing these, not having any issues.

**Comment by matthew-mcallister** [comment]
Is this waiting on something?

**Comment by Gaerzi** [comment]
I don't think so. Graf has been busy cleaning through the old bug and feature suggestion forums, though, that may be why stuff on GitHub is on hold right now.

**Comment by matthew-mcallister** [comment]
OK. Sorry, not trying to rush; I just saw some other PRs get merged in the last week and got to wondering.

File: src/zstring.cpp [hunk 1]
```diff
@@ -393,7 +393,6 @@ void FString::Remove(size_t index, size_t remlen)
 		}
 		else
 		{
-			remlen = Len() - remlen < remlen ? Len() - remlen : remlen;
```
Comment (edward-san) [inline src/zstring.cpp]: Could you explain why was this removed?

**Comment by coelckers** [comment]
That code looks fishy to begin with.  Shouldn't that line have read:

		remlen = Len() - index < remlen ? Len() - index : remlen;

File: src/zstring.cpp (hunk 1 above)
Comment (matthew-mcallister) [inline src/zstring.cpp]: I think @coelckers has it right. I didn't know what to make of the line, but I never tried to delete out of string bounds, so it didn't matter removing it. I believe the console is currently the only code to use this method, so the bug wasn't noticed until now.
